pages/upgrade-guide/4.0.md

```diff
@@ -0,0 +1,186 @@
+
+Instead, jQuery will now only add `px` to a limited set of properties that are known to require it. This change may affect existing code that relies on the old behavior.
+
+### Data
+
+#### Important fix: `Object.prototype` pollution
+
+jQuery 4.0 includes a fix to ensure event and data keys matching `Object.prototype` properties are supported without overriding the native properties. This change is not expected to affect existing code unless it specifically used native property names as event or data keys, which may have caused unexpected behavior anyway.
+
+### Deferred
+
+#### Breaking change: Removed jQuery.Deferred.getStackHook
+
+This has been renamed since 3.7.0 to jQuery.Deferred.getErrorHook. It is used to pass the original error to jQuery.Deferred.exceptionHook. Previously, we recommended passing the stack but that didn't play well with source maps.
+
+### Event
+
+#### Breaking change: Stop shimming focusin & focusout events
+
+jQuery 4.0 no longer shims the `focusin` and `focusout` events. This only affects code that relies on the order of these events as they relate to `focus` and `blur`. This change is expected to affect very few users.
+
+#### Breaking change: Avoid collisions between jQuery.event.special & Object.prototype
+
+This may affect code calling `hasOwnProperty` on `jQuery.event.special`, which should  be rare.
+
+### Manipulation
+
+#### Important feature: Add support for Trusted Types
+
+jQuery 4.0 adds support for [Trusted Types](https://developer.mozilla.org/en-US/docs/Web/API/Trusted_Types_API) for all manipulation methods. This change is not expected to affect existing code, but it is an important feature for security.
+
+#### Potentially breaking change: Avoid concatenating strings in buildFragment
+
+This change was needed in order to add support for [Trusted Types](https://developer.mozilla.org/en-US/docs/Web/API/Trusted_Types_API). It's not entirely clear what real use cases might be affected by this change, but we're listing it here just in case.
+
+#### Important fix: Don't remove HTML comments from scripts
+
+Browsers can now handle this without jQuery's help. This change is not expected to affect existing code, but is an important fix because jQuery's logic removing those comments was not correctt in 100% of cases.
+
+### Selector
+
+#### Breaking change: Drop support for legacy pseudos
+
+References in Sizzle wiki: https://github.com/jquery/sizzle/wiki#-backwards-compatible-plugins-for-pseudos-with-arguments
+
+#### Important fix: Backport jQuery selection context logic to selector-native
+
+jQuery has long had a minimal selector engine that could be used in place of Sizzle. This engine was missing some logic that Sizzle had for handling the context of a selection. This change backports that logic to the native selector engine, which may affect code that relied on the old behavior. For example, `$div.find( "> *" )` will no longer throw an error when using the native selector engine.
```

Comment:
`$div.find( "> *" )` is an important case to mention, but previously it'd throw so this in itself is not breaking. What's breaking is the filtering logic, can we mention a case for this? The most simple would probably be mentioning that with HTML:
```html
<div>
	<span></span>
</div>
```
and `$div` being a jQuery object containing this `div`, `$div.find( "div span" )` would now return 0 results.